client/src/app/pages/assessment-management/assessment-settings/assessment-settings-page.tsx

```diff
@@ -414,18 +423,24 @@ const AssessmentSettings: React.FC = () => {
         title={t("dialog.title.delete", {
           what: t("terms.questionnaire").toLowerCase(),
         })}
+        inProgress={questionnaireToDelete?.name !== questionnaireNameToDelete}
         isOpen={!!questionnaireToDelete}
         titleIconVariant={"warning"}
-        message={t("dialog.message.delete")}
+        message={
+          <DeleteQuestionnaireMessage
+            questionnaireNameToDelete={questionnaireNameToDelete}
+            setQuestionnaireNameToDelete={setQuestionnaireNameToDelete}
+          />
+        }
         confirmBtnVariant={ButtonVariant.danger}
         confirmBtnLabel={t("actions.delete")}
         cancelBtnLabel={t("actions.cancel")}
-        onCancel={() => setQuestionnaireToDelete(null)}
-        onClose={() => setQuestionnaireToDelete(null)}
+        onCancel={() => handleDeleteQuestionnaireClose()}
```

Comment:
all we did before was `setQuestionnaireToDelete(null)` now we run `setQuestionnaireNameToDelete("")` as well. Since I need it in 3 different places in this file I wrote a handler function to avoid code duplications